apps/src/javalab/javabuilderConnection.js

```diff
@@ -0,0 +1,72 @@
+/* globals dashboard */
+// Creates and maintains a websocket connection with javabuilder while a user's code is running.
+export default class JavabuilderConnection {
+  constructor(channelId, javabuilderUrl, onMessage) {
+    this.channelId = channelId;
+    this.javabuilderUrl = javabuilderUrl;
+    this.onOutputMessage = onMessage;
+  }
+
+  // Get the access token to connect to javabuilder and then open the websocket connection.
+  // The token prevents access to our javabuilder AWS execution environment by un-verified users.
+  connectJavabuilder() {
+    $.ajax({
+      url: '/javabuilder/access_token',
+      type: 'get',
+      data: {
+        projectUrl: dashboard.project.getProjectSourcesUrl(),
+        channelId: this.channelId,
```

Comment:
@molly-moen I missed this when looking at the `access_token` [controller action](https://github.com/code-dot-org/code-dot-org/blob/c0b360ad89e6171da78358a6ed5d55fbc91a0a05/dashboard/app/controllers/javabuilder_sessions_controller.rb#L12) ... that logic may need to be updated to validate that the current user owns the project (channel id), with a call to the `owns_channel?` [helper](https://github.com/code-dot-org/code-dot-org/blob/c0b360ad89e6171da78358a6ed5d55fbc91a0a05/shared/middleware/helpers/storage_id.rb#L170).

Reply:
We'd want students to be able to run another student's project. That way students can share projects with each other and with their teacher. It mirrors the same behavior in our other labs.